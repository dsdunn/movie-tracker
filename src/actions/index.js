export const addMovies = (movies) => ({
  type: "ADD_MOVIES",
  movies
});

export const toggleFav = (id) => {
  type: "TOGGLE_FAV",
  id
};

export const toggleFilter = () => {
  type: "TOGGLE_FILTER"
};