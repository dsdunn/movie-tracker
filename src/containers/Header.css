.user-title {
  background-color: #212121;
  color: #CFD8DC;
  font-size: 2em;
  margin: 0;
  padding: 1em;
  border: 0;
  border-radius: 1em 0 0 1em;
  font-family: sans-serif;
}

.favorites {
  font-size: 1.5em;
  background-color: #006992;
  color: #CFD8DC;
  border: 0;
  font-family: sans-serif;
}

.log-out {
  font-size: 1.5em;
  background-color: #27476E;
  color: #CFD8DC;
  border: 0;
  font-family: sans-serif;
}

.sign-in-sign-up {
  padding: .5em 0;
  display: flex;
  justify-content: flex-end;
  background-color: #9E9E9E;
}

.login {
  background-color: #006992;
  color: #CFD8DC;
  font-size: 2em;
  margin: 0;
  padding: 1em;
  border: 0;
  border-radius: 1em 0 0 1em;
  font-family: sans-serif;
}

.signup {
  background-color: #27476E;
  color: #CFD8DC;
  font-size: 2em;
  margin: 0;
  padding: 1em;
  border: 0;
  font-family: sans-serif;
}