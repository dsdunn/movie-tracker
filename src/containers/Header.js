import React from 'react';
import {Link, NavLink, Route, withRouter } from 'react-router-dom';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import './Header.css';
import SignIn from './SignIn.js';
import SignUp from './SignUp.js';

export const Header = (props) => {
  return (
      <div>
        <Link to='/login' component={SignIn}>Login</Link>
        <Link to='/signup' component={SignUp}>Sign Up</Link>
      </div>
  );
};

const mapStateToProps = () => ({
  user: this.state.userName
});

Header.propTypes = {
  userName: PropTypes.object
};

export default withRouter(connect(mapStateToProps)(Header));