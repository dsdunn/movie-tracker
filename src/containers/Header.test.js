import React from 'react';
import { mount } from 'enzyme';
import {toggleFilter, logOutUser} from '../actions';
import { Header, mapStateToProps, mapDispatchToProps } from './Header.js'



describe('header', () => {
  let wrapper
  let mockToggleFilter
  let mockLogOutUser

  beforeEach(() => {
    mockToggleFilter = jest.fn()
    mockLogOutUser = jest.fn()
    wrapper = mount(<Header 
      user={{name: 'Ben'}}
      
      toggleFilter={mockToggleFilter}
      logOutUser={mockLogOutUser}
      />)

  })

  it('should match the snapshot', () => {
   
    expect(wrapper).toMatchSnapshot()
  })

  it('calls toggleFilter on a click of the favorites button', () => {
    wrapper.find('.favorites').simulate('click')
    expect(mockToggleFilter).toHaveBeenCalled()
  })

  it('calls logOutUser when the logout Button is clicked', () => {
    wrapper.find('.log-out').simulate('click')
    expect(mockLogOutUser).toHaveBeenCalled()
  })
})

describe('mapStateToProps', () => {
  it('returns an object with the expected props', () => {
    const mockState = {
      user: {name: 'coffee', id: 1 },
      showAllMovies: true
    }
    const mappedProps = mapStateToProps(mockState)
    const expected = mockState
    expect(mappedProps).toEqual(expected)
  })
})

describe('mapDispatchToProps', () => {
  let mockDispatch

  beforeEach(() => {
    mockDispatch = jest.fn()
  })

  it('calls dispatch when using logOutUser from mDTP', () => {
    const actionToDispatch = logOutUser()
    const mappedProps = mapDispatchToProps(mockDispatch)
    mappedProps.logOutUser()
    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
  })

  it('calls dispatch when using toggleFilter', () => {
    const actionToDispatch = toggleFilter()
    const mappedProps = mapDispatchToProps(mockDispatch)
    mappedProps.toggleFilter()
    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
  })
})

