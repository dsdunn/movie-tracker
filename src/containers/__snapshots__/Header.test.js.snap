// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`header should match the snapshot 1`] = `
ReactWrapper {
  "length": 1,
  Symbol(enzyme.__unrendered__): <Unknown
    logOutUser={[Function]}
    toggleFilter={[Function]}
    user={
        Object {
            "name": "Ben",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "logOutUser": [Function],
      "toggleFilter": [Function],
      "user": Object {
        "name": "Ben",
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": <div>
        <h4>
                Welcome 
                Ben
                !
        </h4>
        <button
                class="log-out"
        >
                Log Out 
        </button>
        <button
                class="favorites"
        >
                Show All
        </button>
</div>,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h4>
             Welcome 
            Ben
             !
</h4>,
          <button
            className="log-out"
            onClick={[Function]}
>
             Log Out 
</button>,
          <button
            className="favorites"
            onClick={[Function]}
>
            Show All
</button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": <h4>
            Welcome 
            Ben
            !
</h4>,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              " Welcome ",
              "Ben",
              " !",
            ],
          },
          "ref": null,
          "rendered": Array [
            " Welcome ",
            "Ben",
            " !",
          ],
          "type": "h4",
        },
        Object {
          "instance": <button
            class="log-out"
>
            Log Out 
</button>,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": " Log Out ",
            "className": "log-out",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Array [
            " Log Out ",
          ],
          "type": "button",
        },
        Object {
          "instance": <button
            class="favorites"
>
            Show All
</button>,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Show All",
            "className": "favorites",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Array [
            "Show All",
          ],
          "type": "button",
        },
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "logOutUser": [Function],
        "toggleFilter": [Function],
        "user": Object {
          "name": "Ben",
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": <div>
          <h4>
                    Welcome 
                    Ben
                    !
          </h4>
          <button
                    class="log-out"
          >
                    Log Out 
          </button>
          <button
                    class="favorites"
          >
                    Show All
          </button>
</div>,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h4>
               Welcome 
              Ben
               !
</h4>,
            <button
              className="log-out"
              onClick={[Function]}
>
               Log Out 
</button>,
            <button
              className="favorites"
              onClick={[Function]}
>
              Show All
</button>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": <h4>
              Welcome 
              Ben
              !
</h4>,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                " Welcome ",
                "Ben",
                " !",
              ],
            },
            "ref": null,
            "rendered": Array [
              " Welcome ",
              "Ben",
              " !",
            ],
            "type": "h4",
          },
          Object {
            "instance": <button
              class="log-out"
>
              Log Out 
</button>,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": " Log Out ",
              "className": "log-out",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Array [
              " Log Out ",
            ],
            "type": "button",
          },
          Object {
            "instance": <button
              class="favorites"
>
              Show All
</button>,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Show All",
              "className": "favorites",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Array [
              "Show All",
            ],
            "type": "button",
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
